src/dune_rules/stanzas/library_redirect.ml

```diff
@@ -17,7 +22,12 @@ module Local = struct
     end)
 
   let for_lib (lib : Library.t) ~new_public_name ~loc : t =
-    { loc; new_public_name; old_name = lib.name; project = lib.project }
+    let old_name =
+      let lib_name = lib.name
+      and enabled = lib.enabled_if in
+      { lib_name; enabled }
+    in
+    { loc; new_public_name; old_name; project = lib.project }
```

Comment:
Just an idea for the future, but I wonder if `old_name` should be renamed to `lib_info` or `target_info` or something like that.

Reply:
I had it `old_info` in an earlier draft but then I changed it back to avoid polluting the diff.